src/main/java/array/Array1d.java

```diff
@@ -0,0 +1,58 @@
+package array;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * One-dimensional array
+ *
+ * @param <T> - type of values that allow to put into array
+ */
+public class Array1d<T> implements Array<T, Coordinates1d> {
+
+    /**
+     * Constructor
+     *
+     * @param size - size of an array
+     */
+    public Array1d(int size) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Factory method for array creation
+     *
+     * @param size - array's size
+     * @param <T> - type of values that allow to put into array
+     * @return array of the given size
+     */
+    public static <T> Array1d<T> of(int size) {
```

Comment:
why do we need both constructor and factory method in here? Looks a bit like an overengineering to me